zerver/lib/actions.py

```diff
@@ -4578,7 +4579,57 @@ def do_update_message(user_profile: UserProfile, message: Message, topic_name: O
         else:
             event['wildcard_mention_user_ids'] = []
 
-    if topic_name is not None:
+    if new_stream is not None:
+        assert content is None
+        assert message.is_stream_message()
+        assert stream_being_edited is not None
+
+        event["stream_id"] = stream_being_edited.id
+        edit_history_event['prev_stream'] = stream_being_edited.id
+        message.recipient_id = new_stream.recipient_id
+
+        event["new_stream_id"] = new_stream.id
+        event["propagate_mode"] = propagate_mode
+
+        old_topic_name = message.topic_name()
+
+        if topic_name is not None:
+            message.set_topic_name(topic_name)
+            # These fields have legacy field names.
+            event[ORIG_TOPIC] = old_topic_name
+            event[TOPIC_NAME] = topic_name
+            event[TOPIC_LINKS] = bugdown.topic_links(message.sender.realm_id, topic_name)
```

Comment:
Would it be possible to avoid duplicating this code with the main `topic_name is not None` code path rather than doing `elif`? 

You're missing code this needs for the `edit_history_event`, and so I think the only complexity is making `changed_messages`  have `Set` semantics rather than `List` semantics.

Or am I missing something here?

Reply:
Ohhh!! I got it! sure! let me try again

Reply:
I updated that part again, Pls take a look :-)